lib/livebook_web/live/session_live.ex

```diff
@@ -1068,6 +1072,44 @@ defmodule LivebookWeb.SessionLive do
     {:noreply, socket}
   end
 
+  @impl true
+  def handle_event(
+        "ai_helper.request_completion",
+        %{
+          "cell_id" => cell_id,
+          "code_before_selection" => code_before_selection,
+          "code_after_selection" => code_after_selection,
+          "selected_code" => selected_code,
+          "prompt" => prompt
+        },
+        socket
+      ) do
+    parent = self()
+
+    # TODO I'm not handling errors here and haven't thought
+    # much about what happens if the liveview process dies and this
+    # is still running
```

Comment:
Latest LiveView (v0.20) supports assign_async for uses cases precisely like these, so once we update it, you should be able to use it just fine!